lib/authorizer/oauth1.js

```diff
@@ -273,48 +417,19 @@ module.exports = {
             }, {});
         }
 
-        // Generate a list of parameters associated with the signature.
-        signatureParams = this.computeHeader({
-            url: getOAuth1BaseUrl(url),
-            method: request.method,
-            queryParams: request.url.query.filter(function (param) {
-                return !param.disabled;
-            }),
-
-            // Body params only need to be included if they are URL encoded.
-            // http://oauth.net/core/1.0a/#anchor13
-            bodyParams: urlencodedBody ? urlencodedBody.filter(function (param) {
-                return !param.disabled;
-            }) : [],
-
-            helperParams: params
-        });
+        // Don't include body hash as defined in specification
+        // @see: https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#when_to_include
+        if (urlencodedBody || !(params.includeBodyHash && hashAlgo)) {
+            self.addAuthDataToRequest(request, params);
 
-        // The OAuth specification says that we should add parameters in the following order of preference:
-        // 1. Auth Header
-        // 2. Body parameters
-        // 3. Query parameters
-        //
-        // http://oauth.net/core/1.0/#consumer_req_param
-        if (params.addParamsToHeader) {
-            header = oAuth1.getAuthorizationHeader(params.realm, _.map(signatureParams, function (param) {
-                return [param.key, param.value];
-            }), params.disableHeaderEncoding);
-            request.addHeader({
-                key: 'Authorization',
-                value: header,
-                system: true
-            });
-        }
-        else if ((/PUT|POST/).test(request.method) && urlencodedBody) {
-            _.forEach(signatureParams, function (param) {
-                urlencodedBody.add(param);
-            });
-        }
-        else {
-            request.addQueryParams(signatureParams);
+            return done();
         }
 
-        return done();
+        computeBodyHash(request.body, hashAlgo, 'base64', function (bodyHash) {
```

Comment:
isn't this adding body hash irrespective of `includeBodyHash` option?

Reply:
It'll only be executed if `includeBodyHash` is true. Otherwise function will return from if block above.